### PR TITLE
Jonmv/more http url

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review. First commits are from #21999

### CODE REVIEW

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java [hunk 1]
```diff
@@ -1472,7 +1472,7 @@ DocumentId id() {
             return new DocumentId("id:" + requireNonNull(path.get("namespace")) +
                                   ":" + requireNonNull(path.get("documentType")) +
                                   ":" + group.map(Group::docIdPart).orElse("") +
-                                  ":" + requireNonNull(path.getRest()));
+                                  ":" + String.join("/", requireNonNull(path.getRest()).segments())); // :'(
```
Comment (bratseth) [inline vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java:1475]: image

File: vespajlib/src/main/java/com/yahoo/path/Path.java [hunk 2]
```diff
@@ -41,6 +41,11 @@ private Path(List<String> elements, String delimiter) {
         this.delimiter = delimiter;
     }
 
+    /** Creates a new path with the given segments. */
+    public static Path from(List<String> segments) {
```
Comment (bratseth) [inline vespajlib/src/main/java/com/yahoo/path/Path.java:45]: A ... signature would be nice I guess

**Review by bratseth — APPROVED** [review]
This is pretty great now

File: vespajlib/src/main/java/com/yahoo/path/Path.java (hunk 2 above)
Comment (jonmv) [reply vespajlib/src/main/java/com/yahoo/path/Path.java:45]: 😄 😭

**Pull request merged**